sdk/js/src/RequestUtils.ts

```diff
@@ -61,15 +61,23 @@ export async function getWithTimeout(
     timeout: number,
 ): Promise<Response> {
     const controller = new AbortController()
-    const id = setTimeout(() => {
-        controller.abort()
-    }, timeout)
-    const response = await get(url, {
-        ...requestConfig,
-        signal: controller.signal,
-    })
-    clearTimeout(id)
-    return response
+    try {
+        const id = setTimeout(() => {
+            controller.abort()
+        }, timeout)
+        const response = await get(url, {
+            ...requestConfig,
+            signal: controller.signal,
+        })
+        clearTimeout(id)
+        return response
+    } catch (e) {
+        if (controller?.signal?.aborted) {
+            throw new Error('Network connection timed out.')
```

Comment:
possible to make a test for this?

Reply:
I don't think it would be a very effective test, because the controller.signal and controller.abort stuff is part of the fetch api which Im mocking out

Reply:
Ill see if I can add an artificial 5 second delay to the fetch mock and see what happens